### PR TITLE
prisma-adapter updated (#1281)

### PR DESCRIPTION
**Pull request by hatsmaker**
## Description

Removed the `@next-auth/prisma-adapter` and  after installed `@auth/prisma-adapter` as described in the issue's description. Changed the `PrismaAdapter` import from `@next-auth/prisma-adapter` to `@auth/prisma-adapter` in `packages/auth/src/server.ts`
## Related Issue

Closes https://github.com/typehero/typehero/issues/1281

### CODE REVIEW

**Comment by bautistaaa** [comment]
this is a duplicate of https://github.com/typehero/typehero/pull/1282

theres type errors though so whoever resolves first wins :)

File: packages/auth/src/server.ts [hunk 1]
```diff
@@ -104,7 +105,7 @@ export const {
       };
     },
   },
-  adapter: PrismaAdapter(prisma),
+  adapter: PrismaAdapter(prisma) as Adapter,
```
Comment (bautistaaa) [inline packages/auth/src/server.ts:108]: feels weird we have to cast for official adapters. is this from the docs?

Comment (hatsmaker) [reply packages/auth/src/server.ts:108]: There is a weird type mismatch error that occurs without an explicit cast, it seems there are some incompatibilities between `next-auth` and `@auth/prisma-adapter` Adapter types.
Link: https://github.com/nextauthjs/next-auth/issues/7727

Comment (bautistaaa) [reply packages/auth/src/server.ts:108]: this issue seems to suggest the type error is supposed to be resolved in 5. oh well i’ll look deeper later

**Review by bautistaaa — APPROVED** [review]

**Pull request merged**